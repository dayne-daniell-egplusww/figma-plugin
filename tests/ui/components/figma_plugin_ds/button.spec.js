import { shallowMount } from '@vue/test-utils';
import FigmaButton from '~/ui/components/figma_plugin_ds/button.vue';
import FigmaLoadingIcon from '~/ui/components/figma_plugin_ds/loading_icon.vue';

describe('Figma button component', () => {
  let wrapper;
  function createComponent({ propsData = {} } = {}) {
    wrapper = shallowMount(FigmaButton, {
      propsData,
      slots: {
        default: 'test',
      },
    });
  }

  describe('renders', () => {
    it('default component', () => {
      createComponent();
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('component with props', () => {
      createComponent({ propsData: { type: 'submit' } });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it.each`
      variant
      ${'primary'}
      ${'secondary'}
      ${'tertiary'}
    `('component with $variant variant class', ({ variant }) => {
      createComponent({ propsData: { variant } });

      expect(wrapper.classes()).toContain(`button--${variant}`);
    });
  });

  it('emits click event when clicked', () => {
    createComponent();
    wrapper.trigger('click');

    return wrapper.vm.$nextTick().then(() => {
      expect(wrapper.emitted().click).toBeTruthy();
    });
  });

  describe('when prop [loading] is true', () => {
    it('displays loading spinner', () => {
      createComponent({ propsData: { loading: true } });
      expect(wrapper.findComponent(FigmaLoadingIcon).exists()).toBe(true);
    });
  });
});
