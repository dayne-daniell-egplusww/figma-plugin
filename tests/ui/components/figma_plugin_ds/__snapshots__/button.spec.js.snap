// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figma button component renders component with props 1`] = `
"<button class=\\"button button--primary\\" type=\\"submit\\">test
  <!----></button>"
`;

exports[`Figma button component renders default component 1`] = `
"<button class=\\"button button--primary\\">test
  <!----></button>"
`;
